### ISSUE TITLE
Can I run this on Windows?

### ISSUE THREAD

**Issue by tyeeman** [dirkk0/microCMS#1]
If I run generator.py will it work on Windows or do I need to convert doit.sh to a batch file?

**Comment by dirkk0** [comment]
One way to do it would be to convert it to a batch file, yes.
The other way would be to transfer the bash commands to Python and maybe additionally to hardcode the values.

Which way you chose is a matter of wether you are better in batch or in Python.

If you chose to write the batch file, consider to create a pull request.

**Comment by tyeeman** [comment]
Are you still using this on your personal site. I looked at your site's source code and it's using "shards toolkit"?

**Comment by dirkk0** [comment]
yes, I do, although the version I use now is a bit more sophisticated.
But the basic principle still stands.
The shards toolkit is a template. It could be any template, but I liked this one the most.
https://designrevision.com/demo/shards/

**Comment by tyeeman** [comment]
Is this code (in doit.sh) - 

```
for FILE in $SITE/input/*
    do FILENAME=$(basename "$FILE" .md)
echo $FILENAME
```

just adding a .md extension to the files? My files on Windows already have a .md extension so do I really need to do this in my batch file?

**Comment by dirkk0** [comment]
In this version of the script, the filename without extension gets processed, so that the python script takes just the filename, and subsequently reads filename.md and writes filename.html.

**Comment by tyeeman** [comment]
OK, just trying to sort out what my batch has to have. So what does the .md do in this line if the files have no .md extension (but the files in the "input" example folder do have the .md extension, so you can see why I'm a little confused).

```
do FILENAME=$(basename "$FILE" .md)
```

**Comment by dirkk0** [comment]
... actually the .md does nothing, I just tried it. :/

You just need a loop to feed the right parameters into the python script.
If you just have a couple of file, you may not even need a loop, but rather do:

    python generator.py page1.md page1.html ../templates
    python generator.py page2.md page2.html ../templates
    ...

etc.

**Comment by tyeeman** [comment]
OK, I get it now, thanks. Have you ever made blog software/script? How hard to add a blog to this one with a listing of blog posts?

**Comment by dirkk0** [comment]
well, you can make a blog with this.
(actually I wrote several content management systems - that's where the
experience comes from)
The focus with this project was static site generation - no javascript,
just python.

If JavaScript is ok with you, you might consider MarkDeep:
https://casual-effects.com/markdeep/

**Comment by tyeeman** [comment]
Yes, I have been playing with markdeep for a few months. It has quite a few great features.

**Comment by dirkk0** [comment]
Indeed.

I find use cases for both, i.e enhancing markdown text with markdeep javascript, and also render clean HMTL/CSS by converting it with microCMS.

I just found out that one could grab the rendered markdeep with `document.querySelector("body").outerHTML`, so maybe there is a third way.

**Comment by tyeeman** [comment]
OK, for windows guys, just run the following batch file from the example-site folder and it generates the site in the www folder, or change it to the "output' folder, whatever you like. 

```
for %%i in (input\*.md) do (
..\generator.py %%i www\%%~ni.html templates
)
```

**Comment by tyeeman** [comment]
Dirk, I took a look at your curious-electric site and it is very cool. Can you actually build it using all markdown files for the content areas? How would I do that with microCMS?

**Comment by dirkk0** [comment]
Yes, it is created by a beefed up version of MicroCMS which includes images and multi-language support via CSV files. It doesn't really use markdown anymore IIRC but I need to look at it.
I you give me a couple of days I'll update this repo with the most recent version; this was on my list for too long already.

**Comment by tyeeman** [comment]
Great, thanks!

**Comment by dirkk0** [comment]
Hey @tyeeman ,
didn't have time to work much more on this.
I'm not giving up on microCMS, I just wonder what the best options are.
See my thoughts here:
https://curious-electric.com/w/blog/#2019-04-01:choosingtherightcms(forablog)

**Comment by tyeeman** [comment]
@dirkk0 
Funny you should mention markdeep again. I write a newsletter for an outdoor club and tried doing it with Markdeep recently and it turned out pretty nice and simple. Here is the sample. It scales nicely on the phone too! It's using apidoc.css template from the markdeep site as you can see viewing the source code. We use mailchimp for the newsletter and it does allow html to be used directly, so I'm going to export that page to html and try pasting that into mailchimp. 

I don't understand why some people say they turn off javascript. Wouldn't that break all sites that use node, react, etc? It's all the rage nowadays and people want to turn it off because of maybe getting malware or viruses? 

A few months ago I decided to make a static site blog generator using a Windows batch file and it works very nicely ( I only know batch unfortunately). Problem is batch files won't work where I want to run it, github or netlify. Do you know how to translate batch to bash or to the GO language? I really like GO because it's just one exe and has no dependencies once created.

**Comment by dirkk0** [comment]
Nice site!
Regarding batch vs bash: I recommend translating to Python, but I am unaware if Netlify supports that. If you send me the batch file, I could try to translate it to bash. My email is my github handle at gmail dot com.
